core/src/main/java/com/linecorp/armeria/client/endpoint/AbstractEndpointSelector.java

```diff
@@ -115,8 +124,59 @@ public final CompletableFuture<Endpoint> select(ClientRequestContext ctx,
         return listeningFuture;
     }
 
+    /**
+     * Initialize this {@link EndpointSelector} to listen to the new endpoints emitted by the
+     * {@link EndpointGroup}. The new endpoints will be passed to {@link #updateNewEndpoints(List)}.
+     */
+    @UnstableApi
+    protected final void initialize() {
```

Comment:
This method was split from the constructor because if the `EndpointGroup` is initialized already, `updateNewEndpoints()` can access member fields from the subclass while initializing the class.